util/cobrautil/completion.go

```diff
@@ -0,0 +1,62 @@
+package cobrautil
+
+import (
+	"strings"
+
+	"github.com/docker/buildx/bake"
+	"github.com/docker/buildx/builder"
+	"github.com/docker/buildx/store/storeutil"
+	"github.com/docker/cli/cli/command"
+	"github.com/spf13/cobra"
+)
+
+// ValidArgsFn defines a completion func to be returned to fetch completion options
+type ValidArgsFn func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective)
+
+func NoCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return nil, cobra.ShellCompDirectiveNoSpace
+}
+
+func CompleteBakeTargets(files []string) ValidArgsFn {
```

Comment:
Perhaps bit if a nit, but if these were in a package named "completion", then these could be `completion.BakeTargets` (etc) which may a bit shorter, and more descriptive